README.md

```diff
@@ -32,27 +32,179 @@ npm install
+    wJEmkMPr5dBOz6DLEeM3IgKNoBlJPp94WSJkgvwTM64rXw049ZkYenl9jwKlcXEmA1a\
+    4MNWoUElr6eh5k20djMZftCYTPUUPMxZUavcQy+cp6lfKonz6HIDe3+n3VOTOo8uu1a\
+    SVfKQQzR+ZEwSaZQBrdQ==:, \
+  proxy_sig=:NgQsRJwOL/EgoRXdcmHMOLZM+KWqLDsO76CrqoiLH279VJs9Fj6bn4V+pe\
+    rAEUbHBEMFCbl6tucEVgKrU+5IIyDMBI85FExQeuBrNPALczjCdxne6LUoBcWBAk8No\
+    Ryjfd++DXIAjAZcf/hBUXLll+5veI0ynzBRFTZ4v8AbluYODjJlSprYEwUb2ndbFr12\
+    vzgIpy0uTQCslN+3rUUZ+lQWlrILvbR0CIvtGwk2+hE0dTRAG0R3wmlR24mhSqiE5RA\
+    DyoSWQVjVxntp98XHAB6MZE92bbu2a8Uo951Hvah03XHWEk/WiYdq+mt3hwXVPLXlBU\
+    9DWCo2AaYD/rkXtQ==:
+```
+
+You can make those headers by using the functions `createSignatureInputHeader` and `createSignatureHeader`.
+
+`createSignatureInputHeader` takes a Map and turns that Map into a `Signature-Input` header.
+
+example: 
 ```js
-const {createAuthzHeader, createSignatureString} = require('http-signature-header');
+const dict = httpSignatureHeader.createSignatureInputHeader({
+  // here we have 1 signatures for the request
+  signatures: new Map([
+    ['sig1', {coveredContent: ['one']}],
+  ]),
+  // this will add the params to all the signatures
+  params: {alg: 'foo'}
+});
+// this is a valid `Signature-Input` header:
+console.log(dict);
+//sig1=("one");alg="foo"
+```
 
-const requestOptions = {
-  url,
-  method: 'POST',
-  headers
-}
-const includeHeaders = ['expires', 'host', '(request-target)'];
-const plaintext = createSignatureString({includeHeaders, requestOptions});
+example: 
+```js
+const dict = httpSignatureHeader.createSignatureInputHeader({
+  // here we have 2 signatures for the request
+  signatures: new Map([
+    ['sig1', {coveredContent: [{value: 'one'}]}],
+    ['sig2', {coveredContent: [{value: 'two'}]}]
+  ]),
+  // this will add the params to all the signatures
+  params: {alg: 'foo'}
+});
+// this is a valid `Signature-Input` header:
+console.log(dict);
+//sig1=("one");alg="foo", sig2=("two");alg="foo"
+```
+example: 
+```js
+const dict = httpSignatureHeader.createSignatureInputHeader({
+  // here we have 2 signatures for the request
+  signatures: new Map([
+    ['sig1', {coveredContent: [{value: 'one'}], params: {alg: 'foo'}}],
+  // sig2 uses a different signing algorithm `bar`
+    ['sig2', {coveredContent: [{value: 'two'}], params: {alg: 'bar'}}]
+  ]),
+});
+// this is a valid `Signature-Input` header:
+console.log(dict);
+//sig1=("one");alg="foo", sig2=("two");alg="bar"
+```
 
-const data = new TextEncoder().encode(plaintext);
-const signature = base64url.encode(await signer.sign({data}));
+`createSignatureInputString`
 
-const Authorization = createAuthzHeader({
-  includeHeaders,
-  keyId: signer.id,
-  signature
+```js
+const date = new Date().toUTCString();
+const httpMessage = {
+  headers: {date},
+  method: 'GET',
+  url: 'https://example.com',
+};
+const coveredContent = [{value: 'one'}];
+const params = {alg: 'foo'};
+// this turns your coveredContent etc into SF Items
```

Comment:
```suggestion
// this turns the coveredContent etc into SF Items
```
Also, `SF` is used here and throughout (variously as `sf`) without introducing the term.  For my part, I'd prefer that it was 'structured field' throughout.